libcrux-ml-kem/tests/wycheproof.rs

```diff
@@ -0,0 +1,109 @@
+use libcrux_kats::wycheproof::mlkem::schema::*;
+
+use libcrux_ml_kem::{MlKemCiphertext, MlKemPublicKey};
+
+macro_rules! wycheproof_test {
+    ($name:ident, $parameter_set:expr, $module:path) => {
+        mod $name {
+            use super::*;
+
+            #[test]
+            fn wycheproof_keygen_and_decaps() {
+                use $module::*;
+
+                let katfile = MlKemTests::load($parameter_set);
+
+                for test_group in katfile.keygen_and_decaps_tests() {
+                    for test in &test_group.tests {
+                        // generate key pair
+                        let key_pair = generate_key_pair(test.seed.clone());
+
+                        // convert ciphertext
+                        let ciphertext: MlKemCiphertext<_> =
+                            <[u8; _]>::try_from(test.ciphertext.clone()).unwrap().into();
+
+                        // validate keys
+                        assert!(validate_private_key(key_pair.private_key(), &ciphertext));
+                        assert!(validate_public_key(key_pair.public_key()));
+
+                        // compare encapsulation key
+                        assert_eq!(key_pair.public_key().as_ref(), test.encapsulation_key);
+
+                        // decapsulate
+                        let shared_secret_from_decapsulate =
+                            decapsulate(key_pair.private_key(), &ciphertext);
+
+                        // compare shared secret
+                        assert_eq!(shared_secret_from_decapsulate, test.shared_secret.as_ref());
+
+                        // assert result is valid
+                        assert_eq!(test.result, MlKemResult::Valid);
+                    }
+                }
+            }
+            #[test]
+            fn wycheproof_encaps() {
+                use $module::*;
+                let katfile = MlKemTests::load($parameter_set);
+
+                for test_group in katfile.encaps_tests() {
+                    for test in &test_group.tests {
+                        // all tests have an Invalid results and a `ModulusOverflow` flag
```

Comment:
Oh only the negative tests are in here. Interesting. Then we should definitely keep the others around.